docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,320 @@
+### Creating factory data JSON file
+
+### Verifying using the JSON Schema tool
+
+The JSON file that contains factory data can be verified using the [JSON Schema tool](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/nrfconnect/nrfconnect_factory_data.schema). This tool validates the structure and contents of the JSON data. Use this tool to validate the factory data JSON file.
+
+To check the JSON file using JSON Schema tool manually on a Linux machine, complete the following steps:
+
+1. Install the ``php-json-schema`` package:
+```
+$ sudo apt install php-json-schema
+```
+
+2. Run the following command, with ``<path_to_JSON_file>`` and ``<path_to_schema_file>`` replaced with the paths to the JSON file and the Schema file, respectively:
+```
+$ validate-json <path_to_JSON_file> <path_to_schema_file>
+```
+
+The tool returns empty output in case of success.
+
+You can have the JSON file be checked automatically by the Python script during its generation by providing the path to the JSON schema file as an additional argument, which should replace the ``<path_to_schema>`` variable in the following command:
+```
+$ python generate_nrfconnect_chip_factory_data.py --schema <path_to_schema>
+```
+
+> Note: To learn more about JSON scheme, visit [this unofficial JSON Schema tool usage website](https://json-schema.org/understanding-json-schema/).
+
+### Preparing factory data partition on a device
+
+The factory data partition is an area in a device's persistent storage where a factory data set is stored. This area is configured using the [Partition Manager](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html), within which all partitions are declared in the ``pm_static.yml`` file.
+
+To prepare an example that supports factory data, a *pm_static* file must contain a partition called ``factory_data``, whose size should be a multiple of one flash page (for nRF52 and nRF53 SoCs, a single page size equals 4 kB).
+
+See the following code snippet for an example of how to create a factory data partition in the ``pm_static.yml`` file.
+The snippet is based on the ``pm_static.yml`` file from the [Lock application example](../../examples/lock-app/nrfconnect/configuration/nrf52840dk_nrf52840/pm_static_dfu.yml) and uses the nRF52840 DK:
+
+
+```
+...
+mcuboot_primary_app:
+    orig_span: &id002
+        - app
+    span: *id002
+    address: 0x7200
+    size: 0xf3e00
+
+factory_data:
+    address: 0xfb00
+    size: 0x1000
+    region: flash_primary
+
+settings_storage:
+    address: 0xfc000
+    size: 0x4000
+    region: flash_primary
+...
+```
+
+In this example, a ``factory_data`` partition has been placed after the application partition (``mcuboot_primary_app`` is a container to store the application firmware that can be replaced in the Direct Firmware Update process). Its size has been set to 4 kB (``0x1000``), which corresponds to one page of the nRF52840 DK flash memory.
```

Comment:
*Direct* Firmware Update of *Device* Firmware Update? :)

Reply:
device :)